### PR TITLE
Add metering to Val conversions, host function invocation

### PR DESCRIPTION
**Pull request by jayz22**
### What

- Adds a new budget type that covers for simple val conversion. E.g `RawVal` to a non-object `ScVal`, or to a `ScObject::U64`. With a test.
- Add `HostFunction` metering to the dispatcher functions. With a test.
- Add several testing utils functions to make it easy to test budget and metering.
- Refactored the invocation tests code to use `utils` functions.

### Why

Part of the bigger picture of budget metering.

### Known limitations

[TODO or N/A]

### CODE REVIEW

File: soroban-env-host/src/budget.rs [hunk 1]
```diff
@@ -153,13 +167,13 @@ impl BudgetDimension {
     }
 
     pub fn is_over_budget(&self) -> bool {
+        if self.limit == u64::MAX {
+            return false;
+        }
```
Comment (graydon) [inline soroban-env-host/src/budget.rs:172]:
```suggestion
```

Comment (graydon) [reply soroban-env-host/src/budget.rs:172]: This is redundant with the existing condition below.

**Review by graydon — CHANGES_REQUESTED** [review]
Looks fine except for the one suggested block to remove.

File: soroban-env-host/src/budget.rs (hunk 1 above)
Comment (jayz22) [reply soroban-env-host/src/budget.rs]: Thanks, fixed.

**Pull request merged**